spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/audit/AuditEvent.java

```diff
@@ -85,10 +87,9 @@ public AuditEvent(String principal, String type, String... data) {
 	public AuditEvent(Date timestamp, String principal, String type,
 			Map<String, Object> data) {
 		Assert.notNull(timestamp, "Timestamp must not be null");
-		Assert.notNull(principal, "Principal must not be null");
 		Assert.notNull(type, "Type must not be null");
 		this.timestamp = timestamp;
-		this.principal = principal;
+		this.principal = principal == null ? StringUtils.EMPTY : principal;
```

Comment:
That check should be reversed: we assert that it's not null, the `else` part dealing with the case when it is.